### PR TITLE
Update data_types.md

### LINKED ISSUE #210: Include section in documentation on data models/data types
It would be useful to have a section in the documentation explaining how we represent the various types of data in the pipeline. For example, an explanation of what a label map is and how that is used to represent segmentations; an explanation of what an xarray is (#209) and how we use that to represent image data; and an explanation of our pandas arrays and how we use that to represent extract information.

### PR DESCRIPTION
**Pull request by ngreenwald**
**If you haven't already, please read through our [contributing guidelines](https://github.com/angelolab/ark-analysis/tree/master/docs/contributing.md) before opening your PR**

**What is the purpose of this PR?**
Closes #210. Provides a description for the key data types in the repo. 

**How did you implement your changes**
First pass for distance matrix to see what you guys think. Does this format seem like it makes sense? Any modifications to how we're linking to documentation, other stuff that should be here, etc?

### CODE REVIEW

**Review by alex-l-kong — APPROVED** [review]
Looks awesome so far!

**Comment by ngreenwald** [comment]
Alright, let me know what you guys think

**Review by alex-l-kong — APPROVED** [review]
Looks great! Definitely will help to give a high level overview of what each of these are.

**Review by ackagel — APPROVED** [review]
Looks great

File: docs/_rtd/data_types.md [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+Used by: [calc_dist_matrix](https://ark-analysis.readthedocs.io/en/latest/_markdown/ark.utils.html#ark.utils.spatial_analysis_utils.calc_dist_matrix)  
+Shape: 4D matrix of fov x num_cells x num_cells x compartment  
+
+Description: This is how segmentation predictions are represented. Each cell is assinged a unique integer value. All of the pixels belonging to that cell have the same value. e.g. all pixels belonging to cell 1 have a value of 1, all pixels belonging to cell 5 have a value of 5, etc. This data structure holds the unique set of labels for each FOV on the first axis. IF there are multiple labels per FOV, for example nuclear and whole-cell, these are stored on the last axis
+
+---
+
+Name: cell table  
+Type: pandas.DataFrame  
+Created by: [generate_cell_data](https://ark-analysis.readthedocs.io/en/latest/_markdown/ark.segmentation.html#ark.segmentation.marker_quantification.generate_cell_data)
```
Comment (ngreenwald) [inline docs/_rtd/data_types.md:19]:
```suggestion
Created by: [generate_cell_table](https://ark-analysis.readthedocs.io/en/latest/_markdown/ark.segmentation.html#ark.segmentation.marker_quantification.generate_cell_table)   
```

**Pull request merged**